mir_eval/separation.py

```diff
@@ -161,28 +158,137 @@ def bss_eval_sources(reference_sources, estimated_sources):
 
     nsrc = estimated_sources.shape[0]
 
-    # compute criteria for all possible pair matches
-    sdr = np.empty((nsrc, nsrc))
-    sir = np.empty((nsrc, nsrc))
-    sar = np.empty((nsrc, nsrc))
-    for jest in range(nsrc):
-        for jtrue in range(nsrc):
+    # does user desire permutations?
+    if compute_permutation:
+        # compute criteria for all possible pair matches
+        sdr = np.empty((nsrc, nsrc))
+        sir = np.empty((nsrc, nsrc))
+        sar = np.empty((nsrc, nsrc))
+        for jest in range(nsrc):
+            for jtrue in range(nsrc):
+                s_true, e_spat, e_interf, e_artif = \
+                    _bss_decomp_mtifilt(reference_sources,
+                                        estimated_sources[jest],
+                                        jtrue, 512)
+                sdr[jest, jtrue], sir[jest, jtrue], sar[jest, jtrue] = \
+                    _bss_source_crit(s_true, e_spat, e_interf, e_artif)
+    else:
+        # compute criteria for only the simple correspondence
+        # (estimate 1 is estimate corresponding to reference source 1, etc.)
+        sdr = np.empty(nsrc)
+        sir = np.empty(nsrc)
+        sar = np.empty(nsrc)
+        for j in range(nsrc):
             s_true, e_spat, e_interf, e_artif = \
                 _bss_decomp_mtifilt(reference_sources,
-                                    estimated_sources[jest],
-                                    jtrue, 512)
-            sdr[jest, jtrue], sir[jest, jtrue], sar[jest, jtrue] = \
+                                    estimated_sources[j],
+                                    j, 512)
+            sdr[j], sir[j], sar[j] = \
                 _bss_source_crit(s_true, e_spat, e_interf, e_artif)
 
-    # select the best ordering
-    perms = list(itertools.permutations(range(nsrc)))
-    mean_sir = np.empty(len(perms))
-    dum = np.arange(nsrc)
-    for (i, perm) in enumerate(perms):
-        mean_sir[i] = np.mean(sir[perm, dum])
-    popt = perms[np.argmax(mean_sir)]
-    idx = (popt, dum)
-    return (sdr[idx], sir[idx], sar[idx], popt)
+    # does user desire permutations?
+    if compute_permutation:
```

Comment:
Can we collapse these two if-statement blocks into one?  E.g. only have one `if compute_permutation`.  I think we can also simplify the logic a bit to avoid most of the special-casing, but we can table that.